### PR TITLE
bug:ensure auctions detail formatted start date time works with live auction integrations

### PR DESCRIPTION
**Pull request by nspinazz89**
This PR addresses an [issue brought up in slack](https://artsy.slack.com/archives/C01PBEBRB0V/p1647482514904939) where the date in force was displaying invalid date to invalid date. This was caused by auctionsDetailFormattedStartDateTime not accounting for live sale integrations.

### CODE REVIEW

**Comment by mzikherman** [comment]
Let's take this opportunity to improve the schema, and move those `auctionDetails` prefixed fields into their own `auctionDetails` object on the `Sale` type.

File: src/lib/date.ts [hunk 1]
```diff
@@ -261,11 +261,21 @@ export function cascadingFormattedStartDateTime(
...
+  endedAt,
+  liveStartAt
+) {
   const thisMoment = moment.tz(moment(), "GMT")
   const saleStartMoment = moment.tz(startAt, "GMT")
   const lotsClosingMoment = moment.tz(endAt, "GMT")
   const saleEndMoment = moment.tz(endedAt, "GMT")
+  const liveStartMoment = moment.tz(liveStartAt, "GMT")
+
+  // return `${saleStartMoment.format("MMM D, YYYY")} • ${saleStartMoment.format(
```
Comment (mzikherman) [inline src/lib/date.ts:276]: possible stray comment?

Comment (nspinazz89) [reply src/lib/date.ts:276]: whoops yeah that was from my debugging

File: src/schema/v2/sale/index.ts [hunk 2]
```diff
@@ -236,30 +236,53 @@ export const SaleType = new GraphQLObjectType<any, ResolverContext>({
...
-              cascading_end_time_interval / 60
-            } minute intervals`
+      cascadingEndTime: {
+        type: new GraphQLObjectType({
+          name: "SaleCascadingEndTime",
+          fields: {
+            formattedStartDateTime: {
+              type: GraphQLString,
+              description:
+                "A more granular formatted description of when the auction starts or ends or if it has ended",
```
Comment (annacarey) [inline src/schema/v2/sale/index.ts:246]: super minor but i think this should be "starts or ends ~~or~~ if it has ended"

File: src/schema/v2/sale/index.ts [hunk 3]
```diff
@@ -236,30 +236,53 @@ export const SaleType = new GraphQLObjectType<any, ResolverContext>({
...
+              resolve: (
+                { start_at, end_at, ended_at, live_start_at },
+                _options
+              ) => {
+                return auctionsDetailFormattedStartDateTime(
```
Comment (annacarey) [inline src/schema/v2/sale/index.ts:251]: Maybe change the variable name?

File: src/schema/v2/sale/index.ts [hunk 4]
```diff
@@ -236,30 +236,53 @@ export const SaleType = new GraphQLObjectType<any, ResolverContext>({
...
+                } else {
+                  return `Lots close at ${
+                    cascading_end_time_interval / 60
+                  }-minute intervals`
+                }
+              },
+            },
+          },
+        }),
+        resolve: (sale) => {
```
Comment (annacarey) [inline src/schema/v2/sale/index.ts:279]: I **believe** you can do: 

```
resolve: ({start_at, end_at, ended_at, live_start_at, cascading_end_time_interval}) => {
 return {
            start_at,
            end_at,
            ended_at,
            live_start_at,
            cascading_end_time_interval
          }

Comment (annacarey) [reply src/schema/v2/sale/index.ts:279]: could be wrong

File: src/lib/date.ts [hunk 5]
```diff
@@ -261,11 +261,17 @@ export function cascadingFormattedStartDateTime(
...
+  endedAt,
+  liveStartAt
+) {
   const thisMoment = moment.tz(moment(), "GMT")
   const saleStartMoment = moment.tz(startAt, "GMT")
   const lotsClosingMoment = moment.tz(endAt, "GMT")
   const saleEndMoment = moment.tz(endedAt, "GMT")
+  const liveStartMoment = moment.tz(liveStartAt, "GMT")
 
   if (!!endedAt)
```
Comment (annacarey) [inline src/lib/date.ts:276]: do you need the double `!`?

**Review by annacarey — APPROVED** [review]
This looks great! Way to see it through. All of my comments are very minor so feel free to merge when ready.

File: src/lib/date.ts (hunk 5 above)
Comment (nspinazz89) [reply src/lib/date.ts:276]: yeah that forces it into a boolean I believe.

**Pull request merged**